src/lib/utils/tiptap/resourceReferenceMark.ts

```diff
@@ -16,14 +16,20 @@ export const resourceReferenceMark = Mark.create<ResourceReferenceOptions>({
     renderHTML({ HTMLAttributes }) {
         const associatedResourceInfo = this.options.availableAssociatedResources?.find(
             (ar) =>
-                ar.externalId === HTMLAttributes.resourceId || ar.resourceId.toString() === HTMLAttributes.resourceId
+                (ar.mediaType === MediaType.Text || !ar.mediaType) &&
+                (ar.externalId === HTMLAttributes.resourceId || ar.resourceId.toString() === HTMLAttributes.resourceId)
         );
+        const audioId = this.options.availableAssociatedResources?.find(
+            (ar) => ar.mediaType === MediaType.Audio && ar.resourceId === associatedResourceInfo?.resourceId
+        )?.contentId;
         if (associatedResourceInfo) {
             return [
                 'button',
                 {
                     class: 'btn-link',
-                    onClick: `onResourceReferenceClick('${this.options.tab}', ${associatedResourceInfo.contentId})`,
+                    onClick: `onResourceReferenceClick('${this.options.tab}', ${associatedResourceInfo.contentId}, ${
+                        audioId ?? 'undefined'
```

Comment:
Is this supposed to be `undefined` the keyword, rather than string?

Reply:
Since it's being interpolated into the string it will end up like `onResourceReferenceClick(..., undefined)`